koku/api/report/queries.py

```diff
@@ -132,11 +132,29 @@ def _get_search_filter(self, filters):
         # Update filters that specifiy and or or in the query parameter
         and_composed_filters = self._set_operator_specified_filters('and')
         or_composed_filters = self._set_operator_specified_filters('or')
+        multi_field_or_composed_filters = self._set_or_filters()
         composed_filters = filters.compose()
         composed_filters = composed_filters & and_composed_filters & or_composed_filters
+        if multi_field_or_composed_filters:
+            composed_filters = composed_filters & multi_field_or_composed_filters
```

Comment:
This section seems like it's over-complex. 

We've got a variable for the AND filters, a second variable for the OR filters, and now a new, third variable for a different set of OR filters assembled with new and different criteria.

For the scope of this PR, this is fine. But, to reiterate the point in a previous comment, I think we should spawn a new issue to re-think our filtering aggregating mechanisms.

Reply:
Couldn't agree more.